### PR TITLE
Bare-bones api

### PR DESCRIPTION
**Pull request by LucasTJerez**
This pull request introduces an api that allows for restaurant and customer accounts to login in, retrieve data by id, and some more functions specifically for customers (like getting orders). I added the bare minimum amount of functions that Customer objects need to instantiate themselves

### CODE REVIEW

File: backend/Database.py [hunk 1]
```diff
@@ -167,6 +167,14 @@ def removeById(self, id):
 
         self.data["ids"].remove(id)
 
+
+    def getById(self, id):
+        potentialComponent = [self.getCustomerById(id), self.getRestaurantById(id), self.getItemById(id), self.getOrderById(id)]
```
Comment (tgomezzzz) [inline backend/Database.py:172]: Do IDs overlap between objects? For example, could there be a customer and an item both with ID 1, or would they need to have completely different IDs?

Comment (tgomezzzz) [reply backend/Database.py:172]: If they do, we might consider changing IDs to be unique among different types of objects. We can assign certain IDs to a particular group (ie, all restaurant IDs start with 4 or something), so we don't have to waste time running getCustomerID(), getItemID(), when we already know what object we're expecting.

Comment (tgomezzzz) [reply backend/Database.py:172]: We can also use strings as IDs, so Customers could have IDs C0, C1, C2, Restaurants could have R0, R1, and so on

File: backend/Database.py [hunk 2]
```diff
@@ -199,6 +207,13 @@ def getOrderById(self, id):
             if order["id"] == id:
                 return order
 
+    def getCustomerOrders(self, custId):
+        orders = []
+        for order in self.data["orders"]:
```
Comment (tgomezzzz) [inline backend/Database.py:212]: Maybe we could add a list of order IDs to each customer's dictionary, so we don't have to iterate over every order in the database

File: backend/api.py [hunk 3]
```diff
@@ -1,28 +1,88 @@
...
+
+    if username in loggedInUsers:
+        return jsonify("already logged in")
+
+    loggedInUsers.add(username)
+    newKey = generateKey()
+    loginKeys.add(newKey)
+    userId = db.data['logins'][username]['id']
+    
+    return({'key':newKey, 'id':userId})
```
Comment (tgomezzzz) [inline backend/api.py:35]: What is the userId?

**Review by tgomezzzz — APPROVED** [review]

File: backend/Database.py (hunk 2 above)
Comment (LucasTJerez) [reply backend/Database.py:212]: Yup, that's a good idea

**Pull request merged**